src/test/ui/closures/coerce-unsafe-to-closure.stderr

```diff
@@ -7,6 +7,7 @@ LL |     let x: Option<&[u8]> = Some("foo").map(std::mem::transmute);
    |                                        required by a bound introduced by this call
    |
    = help: the trait `FnOnce<(&str,)>` is not implemented for `unsafe extern "rust-intrinsic" fn(_) -> _ {transmute::<_, _>}`
+   = note: unsafe functions must be wrapped in closures: `|| unsafe { /* code */ }`
```

Comment:
Can we can add a `message` to the `on_unimplemented` attribute explaining _why_ `unsafe` functions don't implement the `Fn` family of traits?

Reply:
I would like to do that, but how would I do it in a one-liner? The truth is basically this:

> `unsafe` functions do not implement the `Fn` family of traits, because you need to discharge the proof obligation on the unsafe functions in order to call it safely, and since proof obligations are not a part of the type system, a generic function invoking a trait cannot possibly know what the proof obligation is

Yeah, I'm not putting that in an error message.

Reply:
Well, "`unsafe` functions do not implement `{Trait}` because they cannot be called generically as safe functions" or something shorter might do. I don't particularly giving the advice to just add an `unsafe` block without explaining _why_.

Reply:
Is this better?

```
error[E0277]: expected a `FnOnce<(&isize,)>` closure, found `unsafe fn(isize) -> isize {square}`
  --> $DIR/unboxed-closures-wrong-arg-type-extern-fn.rs:31:26
   |
LL |     let z = call_it_once(square, 22);
   |             ------------ ^^^^^^ call the function in a closure: `|| unsafe { /* code */ }`
   |             |
   |             required by a bound introduced by this call
   |
   = help: the trait `for<'r> FnOnce<(&'r isize,)>` is not implemented for `unsafe fn(isize) -> isize {square}`
   = note: unsafe function cannot be called generically without an unsafe block
```